src/app/parcels/ParcelsPage.tsx

```diff
@@ -147,18 +153,68 @@ const ParcelsPage: React.FC<{}> = () => {
     }, [packingDateRange]);
 
     useEffect(() => {
-        // This requires that the DB parcels table has Realtime turned on
+        // This requires that both the DB parcels and events tables have Realtime turned on
+        const loadParcelTableData = async (): Promise<void> => {
+            if (timer.current) {
+                clearTimeout(timer.current);
+                timer.current = null;
+            }
+
+            setIsLoading(true);
+            timer.current = setTimeout(async () => {
+                setTableData(await fetchAndFormatParcelTablesData(packingDateRange));
+                setIsLoading(false);
+            }, 500);
+        };
+
         const subscriptionChannel = supabase
             .channel("parcels-table-changes")
-            .on("postgres_changes", { event: "*", schema: "public", table: "parcels" }, async () =>
-                setTableData(await fetchAndFormatParcelTablesData(packingDateRange))
+            .on(
+                "postgres_changes",
+                { event: "*", schema: "public", table: "parcels" },
+                async () => {
+                    await loadParcelTableData();
+                }
             )
+            .on("postgres_changes", { event: "*", schema: "public", table: "events" }, async () => {
+                await loadParcelTableData();
+            })
             .subscribe();
 
         return () => {
             supabase.removeChannel(subscriptionChannel);
         };
-    }, [packingDateRange]);
+    }, [packingDateRange, isLoading]);
+
+    useEffect(() => {
+        setSelectedRowIndices([]);
+    }, [tableData]);
+
+    const selectOrDeselectRow = (rowIndex: number): void => {
+        setSelectedRowIndices((currentIndices) => {
+            if (currentIndices.includes(rowIndex)) {
+                return currentIndices.filter((index) => index !== rowIndex);
+            }
+            return currentIndices.concat([rowIndex]);
+        });
+    };
+
+    const toggleAllCheckBox = (isAllCheckBoxSelected: boolean): void => {
```

Comment:
Do you need this parameter at all? Could we just use the `isAllCheckBoxSelected` state?
```suggestion
    const toggleAllCheckBox = (): void => {
```